src/include/migraphx/op/dequantizelinear.hpp

```diff
@@ -0,0 +1,82 @@
+#ifndef MIGRAPHX_GUARD_OPERATORS_DEQUANTIZE_LINEAR_HPP
+#define MIGRAPHX_GUARD_OPERATORS_DEQUANTIZE_LINEAR_HPP
+
+#include <array>
+#include <migraphx/op/common.hpp>
+#include <migraphx/operation.hpp>
+#include <migraphx/check_shapes.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/streamutils.hpp>
+#include <migraphx/literal.hpp>
+#include <migraphx/config.hpp>
+#include <migraphx/par_for.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/op/normalize_attribute.hpp>
+#include <migraphx/tune_axis.hpp>
+#include <cmath>
+#include <utility>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace op {
+
+struct dequantizelinear
+{
+    int axis = 1;
+
+    template <class Self, class F>
+    static auto reflect(Self& self, F f)
+    {
+        return pack(f(self.axis, "axis"));
+    }
+
+    std::string name() const { return "dequantizelinear"; }
+    shape compute_shape(std::vector<shape> inputs) const
+    {
+        return {shape::float_type, inputs[0].lens(), inputs[0].strides()};
+    }
+
+    argument compute(const shape& output_shape, std::vector<argument> args) const
```

Comment:
The compute function can be something like:

```
    argument compute(const shape& output_shape, std::vector<argument> args) const
    {
        std::vector<int8_t> zeros(output_shape.bytes(), 0);
        argument zero_point{output_shape, zeros.data()};
        if (args.size() == 3)
        {
            zero_point = args.at(2);
        }

        argument result{output_shape};
        visit_all(result, args.at(1))([&](auto output, auto scale) {
            visit_all(args.at(0), zero_point)([&](auto x, auto zp) {
                par_for(output_shape.elements(), [&](auto i) {
                    output[i] = static_cast<float>((x[i] - zp[i])) * scale[i];
                });                
            });
        });

        return result;
    }
```
if all inputs are broadcasted.

Reply:
That is not correct, `output` and `zero_point` may not have the same type.  The `zero_point` should match the type of the input. Also, the calculation should use `int64_t` and `double`.

Reply:
The implementation does NOT require output and zero_point have the same type. It requires:
x and zero_point have the same type. 
according to the onnx specification, output and scale have the same type. That is why I wrote like this.